File: ec2-instance/libs/vpcStack.py
from aws_cdk import (
    Stack
)

from aws_cdk.aws_ec2 import (
    Vpc,
    IpAddresses,
    SubnetConfiguration,
    SubnetType,
    SecurityGroup,
    Peer,
    Port
)

from constructs import Construct

class VpcStack(Stack):
        
    def __init__(self, scope: Construct, id: str, cidr: str, creator: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)
        
        # Create a VPC with public subnets in 2 AZs
        self.vpc = Vpc(self, f"VPC-{creator}",
            ip_addresses=IpAddresses.cidr(f"10.{int(cidr)}.0.0/16"),
            nat_gateways=0,
            subnet_configuration=[
                SubnetConfiguration(
                    cidr_mask=24,
                    name="PublicSubnet",
                    subnet_type=SubnetType.PUBLIC,
                    map_public_ip_on_launch=True
                )
            ],
            max_azs=2
        )
        
        # Create a security group for SSH
        self.sshSecurityGroup = SecurityGroup(self, f"SSHSecurityGroup-{creator}",
            vpc = self.vpc,
            description="Security Group for SSH",
            allow_all_outbound=True
        )

        # Allow SSH inbound traffic on TCP port 22
        self.sshSecurityGroup.add_ingress_rule(Peer.any_ipv4(),Port.tcp(22))

File: typescript/ec2-instance/src/libs/vpcStack.ts
import { Stack, StackProps } from 'aws-cdk-lib';
import {
  SecurityGroup,
  Peer,
  Port,
  SubnetType,
  Vpc,
  IpAddresses
} from 'aws-cdk-lib/aws-ec2';
import { Construct } from 'constructs';

export interface VpcProps extends StackProps {
  cidr: number;
  creator: string;
}

export class VpcStack extends Stack {
  public readonly sshSecurityGroup: SecurityGroup;
  public readonly vpc: Vpc;

  constructor(scope: Construct, id: string, props:VpcProps) {
    super(scope, id);


    // Create a VPC with public subnets in 2 AZs
    this.vpc = new Vpc(this, `VPC-${props.creator}`, {
      ipAddresses: IpAddresses.cidr(`10.${props.cidr}.0.0/16`),
      natGateways: 0,
      subnetConfiguration: [
        {
          cidrMask: 24,
          name: 'PublicSubnet',
          subnetType: SubnetType.PUBLIC,
          mapPublicIpOnLaunch: true,
        },
      ],
      maxAzs: 2,
    });

    // Create a security group for SSH
    this.sshSecurityGroup = new SecurityGroup(this, `SSHSecurityGroup-${props.creator}`, {
      vpc: this.vpc,
      description: 'Security Group for SSH',
      allowAllOutbound: true,
    });

    // Allow SSH inbound traffic on TCP port 22
    this.sshSecurityGroup.addIngressRule(Peer.anyIpv4(), Port.tcp(22));
  }
}
